SpatialGDK/Source/SpatialGDKEditor/Public/SpatialOSLoadBalancingStrategies.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "UObject/Object.h"
+#include "Serialization/JsonWriter.h"
+
+#include "SpatialOSLoadBalancingStrategies.generated.h"
+
+UCLASS(Abstract)
+class SPATIALGDKEDITOR_API USpatialOSAbstractLoadBalancingStrategy : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual bool WriteToConfiguration(TSharedRef<TJsonWriter<>> Writer) const
+	{
+		check(0); return false;
+	}
+
+	virtual int32 GetNumberOfWorkersForPIE() const
+	{
+		check(0); return false;
+	}
+};
+
+UCLASS()
+class SPATIALGDKEDITOR_API USpatialOSSingleWorkerStrategy : public USpatialOSAbstractLoadBalancingStrategy
+{
+	GENERATED_BODY()
+
+public:
+
+	USpatialOSSingleWorkerStrategy();
+
+	bool WriteToConfiguration(TSharedRef<TJsonWriter<>> Writer) const override;
+
+	int32 GetNumberOfWorkersForPIE() const override;
+};
+
+UCLASS(EditInlineNew)
+class SPATIALGDKEDITOR_API USpatialOSGridLoadBalancingStrategy : public USpatialOSAbstractLoadBalancingStrategy
+{
+	GENERATED_BODY()
+
+public:
+
+	USpatialOSGridLoadBalancingStrategy();
+
+	/** Number of columns in the rectangle grid load balancing config. */
+	UPROPERTY(EditAnywhere, meta = (DisplayName = "Rectangle grid column count", ClampMin = "1", UIMin = "1"))
+	int32 Columns;
+	
+	/** Number of rows in the rectangle grid load balancing config. */
+	UPROPERTY(EditAnywhere, meta = (DisplayName = "Rectangle grid row count", ClampMin = "1", UIMin = "1"))
+	int32 Rows;
+	
+	bool WriteToConfiguration(TSharedRef<TJsonWriter<>> Writer) const override;
+
+	int32 GetNumberOfWorkersForPIE() const override;
```

Comment:
does this still allow us to start up less workers than specified in the column and row fields? cause this was pretty useful afaik for debugging purposes

Reply:
With what I did, that will have to be done with a custom launch configuration file.

Reply:
what would be the flow to switch between different configurations when running PIE

Reply:
Actually no that would not work :/
I'l add a new setting to do it.

Reply:
Done, one can override per worker typer the number of in editor workers to launch